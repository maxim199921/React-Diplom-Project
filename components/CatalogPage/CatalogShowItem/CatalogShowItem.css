.CatalogShowItem {
    margin-top: 220px;
    box-shadow: 0 0 15px 0 rgba(22,35,23,.4);
    margin-bottom: 97px;
}

.CatalogShowItem-img-wrapper {
    min-height: 220px;
    margin-top: 30px;
    margin-bottom: 30px;
    display: flex;
}

.CatalogShowItem-img {
    margin: auto;
    min-height: 220px;
    max-width: 100%;
}

.CatalogShowItem-wrapper {
    margin-top: 55px;
}

.CatalogShowItem-text {
    font-size: 20px;
    font-weight: bold;
    line-height: 25px;
    margin-bottom: 20px;
}

.CatalogShowItem-price {
    color: orange;
    font-size: 22px;
    font-weight: bold;
}

.CatalogShowItem-type,
.CatalogShowItem-model,
.CatalogShowItem-description {
    margin-top: 3px;
}


@media (max-width: 575px) {
    .CatalogShowItem {
        margin-top: 120px;
        box-shadow: none;
    }
    .CatalogShowItem-text {
        display: none;
    }
    .CatalogShowItem-type,
    .CatalogShowItem-model,
    .CatalogShowItem-description,
    .CatalogShowItem-price {
        text-align: center;
        margin-top: 10px;
    }
    .CatalogShowItem-type {
        margin-bottom: 10px;
        font-size: 24px;
        font-weight: bold;
    }
    .CatalogShowItem-img-wrapper {
        margin-bottom: 0;
    }
    .CatalogShowItem-img {
        min-height: 220px;
    }

    .CatalogShowItem-wrapper {
        margin-top: 10px;
    }
}

.CatalogPage-Loader-2 {
    margin-top: 120px;
}