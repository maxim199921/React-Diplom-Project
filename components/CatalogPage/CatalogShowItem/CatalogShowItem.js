import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import './CatalogShowItem.css';
import {loadData} from "../../../redux/ActionCreaters/CatalogActions";

class CatalogShowItem extends React.PureComponent {

    static propTypes = {
        itemId: PropTypes.any,
        isLoaded: PropTypes.bool.isRequired,
        startPriceList: PropTypes.arrayOf(
            PropTypes.shape({
                id: PropTypes.number.isRequired,
                type: PropTypes.string.isRequired,
                img: PropTypes.string.isRequired,
                model: PropTypes.string.isRequired,
                price: PropTypes.string.isRequired,
                description: PropTypes.string.isRequired,
                status: PropTypes.bool.isRequired,
            })
        ),
    };

    state = {
        info: {}
    };

    componentDidMount = () => {
        this.props.dispatch(loadData());
    };
    componentDidUpdate = () => {
        let getItem = this.props.startPriceList.filter((item) => item.id === +this.props.itemId);
        this.setState({info: getItem[0]});
    };

    render() {

        let item = this.props.startPriceList[this.props.itemId - 1];

        return (
            (this.props.isLoaded)?
            <div className="container CatalogShowItem ">
                <div className="row CatalogShowItem-wrapper">
                    <div className="col-sm-4 col-lg-6 CatalogShowItem-img-wrapper">
                        <img className="CatalogShowItem-img" src={item.img||this.state.info.img} />
                    </div>
                    <div className="col-sm-8 col-lg-6">
                        <div className="row CatalogShowItem-wrapper">
                            <div className="col-4 CatalogShowItem-text">Type:</div>
                            <div className="col-sm-8 CatalogShowItem-type">{item.type||this.state.info.type}</div>
                            <div className="col-4 CatalogShowItem-text">Model:</div>
                            <div className="col-sm-8 CatalogShowItem-model">{item.model||this.state.info.model}</div>
                            <div className="col-4 CatalogShowItem-text">Description:</div>
                            <div className="col-sm-8 CatalogShowItem-description">{item.description||this.state.info.description}</div>
                            <div className="col-4 CatalogShowItem-text">Price:</div>
                            <div className="col-sm-8 CatalogShowItem-price">{item.price||this.state.info.price}</div>
                        </div>
                    </div>
                </div>
            </div>
            :
            <div className="CatalogPage-Loader-2 CatalogShowItem">
                <div className="container">
                    <div className="row">
                        <div className="col-12">
                            <h1 className="CatalogPage-Text-Loading">...Loading
                                <div className="loader-container">
                                    <div className="loader-1"/>
                                </div>
                            </h1>
                        </div>
                    </div>
                </div>
            </div>
        );

    }

}

const mapStateToProps = function (state) {
    return {
        startPriceList: state.catalogList.startPriceList,
        isLoaded: state.catalogList.isLoaded,
    };
};


export default connect(mapStateToProps)(CatalogShowItem);

