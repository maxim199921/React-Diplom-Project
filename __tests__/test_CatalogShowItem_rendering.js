import React from 'react';
import renderer from 'react-test-renderer';
import CatalogShowItem from '../components/CatalogPage/CatalogShowItem/CatalogShowItem';
import {HashRouter, hashHistory} from 'react-router-dom';
import {Provider} from 'react-redux';
import configureStore from '../redux/configureStore';
const store = configureStore({});


test('рендеринг одностраничного просмотра каталога', () => {

    const component = renderer.create(
        <Provider store={store}>
            <HashRouter history={hashHistory}>
                <CatalogShowItem/>
            </HashRouter>
        </Provider>

    );

    let componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

});