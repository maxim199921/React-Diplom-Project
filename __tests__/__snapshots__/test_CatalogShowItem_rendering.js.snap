// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`рендеринг одностраничного просмотра каталога 1`] = `
<div
  className="CatalogPage-Loader-2 CatalogShowItem"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <h1
          className="CatalogPage-Text-Loading"
        >
          ...Loading
          <div
            className="loader-container"
          >
            <div
              className="loader-1"
            />
          </div>
        </h1>
      </div>
    </div>
  </div>
</div>
`;
